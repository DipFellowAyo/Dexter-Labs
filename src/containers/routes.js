import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'

import {
  ADMIN_AUTH,
  LANDING_ROUTE,
  FORGOT_PASSWORD_ROUTE,
  LOGIN_ROUTE,
  MERCHANTS_ROUTE,
  NOT_FOUND_ROUTE,
  PASSWORD_PERSONALIZATION_ROUTE,
  PASSWORD_RESET_SUCCESSFULL_ROUTE,
  REGISTER_ROUTE,
  REGISTRATION_SUCCESSFUL_ROUTE,
  TRANSACTIONS_ROUTE,
  DAASHBOARD_ROUTE,
  MESSAGES_ROUTE,
  DSA_DASHBOARD_ROUTE
} from '../utils/routes'

const Landing = React.lazy(() => import('../pages/landing'))
const Login = React.lazy(() => import('../pages/login'))
const Register = React.lazy(() => import('../pages/register'))
const ForgotPassword = React.lazy(() => import('../pages/forgot-password'))
const PasswordResetForgotSuccessful = React.lazy(() =>
  import('../pages/password-reset-forgot-successful')
)
const RegistrationSuccessful = React.lazy(() =>
  import('../pages/registration-successful')
)
const AdminAuth = React.lazy(() => import('../pages/admin-auth'))
const PasswordPersonalization = React.lazy(() =>
  import('../pages/password-personalization')
)
const Dashboard = React.lazy(() => import('../pages/dashboard/index'))
const Merchants = React.lazy(() => import('../pages/dashboard/merchants'))
const Transactions = React.lazy(() => import('../pages/dashboard/transactions'))
const Messages = React.lazy(() => import('../pages/dashboard/messages'))
const NotFound = React.lazy(() => import('../pages/not-found'))
const DSA_Dashboard = React.lazy(() => import('../pages/DSA_dashboard/index'))

const Routes = () => {
  return (
    <Switch>
      <Route exact path={LANDING_ROUTE} component={Landing} />
      <Route exact path={LOGIN_ROUTE} component={Login} />
      <Route exact path={REGISTER_ROUTE} component={Register} />
      <Route exact path={FORGOT_PASSWORD_ROUTE} component={ForgotPassword} />
      <Route
        exact
        path={PASSWORD_RESET_SUCCESSFULL_ROUTE}
        component={PasswordResetForgotSuccessful}
      />
      <Route
        exact
        path={REGISTRATION_SUCCESSFUL_ROUTE}
        component={RegistrationSuccessful}
      />
      <Route exact path={ADMIN_AUTH} component={AdminAuth} />
      <Route
        exact
        path={PASSWORD_PERSONALIZATION_ROUTE}
        component={PasswordPersonalization}
      />
      <Route exact path={DSA_DASHBOARD_ROUTE} component={DSA_Dashboard} />
      <Route exact path={DAASHBOARD_ROUTE} component={Dashboard} />
      <Route exact path={MERCHANTS_ROUTE} component={Merchants} />
      <Route exact path={TRANSACTIONS_ROUTE} component={Transactions} />
      <Route exact path={MESSAGES_ROUTE} component={Messages} />
      <Route exact path={NOT_FOUND_ROUTE} component={NotFound} />
      <Redirect from='*' to={NOT_FOUND_ROUTE} />
    </Switch>
  )
}

export default Routes
