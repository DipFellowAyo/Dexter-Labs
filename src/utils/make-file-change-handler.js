export default function makeFileChangeHandler ({
  key,
  // setPreview,
  setFieldValue,
  setFieldError,
  setFieldTouched,
  maxFileSize = 1,
  mimeErrorMessage = 'Images only please.',
  fileTypeRegex = /^data:image\/(\w+);base64,/,
  fileSizeErrorMessage = `File size cannot be more than ${maxFileSize}MB.`
}) {
  return e => {
    e.persist()

    if (!e.target.files.length) {
      return setFieldValue(key, '')
    }

    setFieldTouched(key, true, false)

    const file = e.target.files[0]

    const fileSize = file.size / 1024 / 1024

    if (fileSize > maxFileSize) {
      e.target.value = ''

      return setFieldError(key, fileSizeErrorMessage)
    }

    const reader = new window.FileReader()

    reader.readAsDataURL(file)

    reader.onload = () => {
      const dataURI = reader.result

      if (!fileTypeRegex.test(dataURI)) {
        return setFieldError(key, mimeErrorMessage)
      }

      // setPreview(dataURI)

      setFieldValue(key, dataURI)
    }

    reader.onerror = () => setFieldValue(key, '')
  }
}
