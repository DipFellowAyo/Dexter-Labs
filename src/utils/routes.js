export const LANDING_ROUTE = '/'

export const LOGIN_ROUTE = '/login'

export const FORGOT_PASSWORD_ROUTE = '/forgot-password'

export const REGISTER_ROUTE = '/register'

export const PASSWORD_RESET_SUCCESSFULL_ROUTE = '/password-reset-successful'

export const REGISTRATION_SUCCESSFUL_ROUTE = '/registration-successful'

export const ADMIN_AUTH = '/admin-auth'

export const PASSWORD_PERSONALIZATION_ROUTE = '/password-personalization'

export const DSA_DASHBOARD_ROUTE = '/dsa-dashboard'

export const DAASHBOARD_ROUTE = '/dashboard'

export const MERCHANTS_ROUTE = '/dashboard/merchants'

export const TRANSACTIONS_ROUTE = '/dashboard/transactions'

export const MESSAGES_ROUTE = '/dashboard/messages'

export const NOT_FOUND_ROUTE = '/not-found'
