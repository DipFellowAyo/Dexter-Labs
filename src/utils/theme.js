import { createTheme } from '@material-ui/core/styles'

const primaryColor = '#6F6F6F'
const secondaryColor = '#ec7e81'

export const theme = createTheme({
  typography: {
    fontFamily: 'Poppins'
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '@font-face': ['Poppins']
      }
    }
  },
  palette: {
    primary: {
      main: primaryColor
    },
    secondary: {
      main: secondaryColor
    }
  }
})
