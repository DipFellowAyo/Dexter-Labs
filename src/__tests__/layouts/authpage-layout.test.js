import React from 'react'
import Box from '@material-ui/core/Box'
import toJson from 'enzyme-to-json'
import { Link } from 'react-router-dom'
import { shallow } from 'enzyme'

import AuthPageLayout from '../../layouts/authpage.layout'
import { LANDING_ROUTE } from '../../utils/routes'

describe('AuthPage Layout', () => {
  let authPageLayout

  beforeEach(() => {
    authPageLayout = shallow(<AuthPageLayout />)
  })
  afterAll(() => jest.resetAllMocks())

  it('should render without crashing', () => {
    expect(authPageLayout).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(toJson(authPageLayout)).toMatchSnapshot()
  })

  it('should render the AuthPage Layout with a prop name `title`', () => {
    expect(authPageLayout.props().title).toBeUndefined()
  })

  it('should render Box component', () => {
    expect(authPageLayout.find(Box)).toBeTruthy()
  })

  it('should render the Box component with a prop name `flexGrow` at node 0', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(0)
        .props().flexGrow
    ).toBe('1')
  })

  it('should render the Box component with a prop name `className` at node 0', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(0)
        .props().className
    ).toBe('makeStyles-root-1')
  })

  it('should render the Box component with a prop name `className` at node 1', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(1)
        .props().className
    ).toBe('makeStyles-gridContainer-2')
  })

  it('should render the Box component with a prop name `className` at node 2', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(2)
        .props().className
    ).toBe('makeStyles-leftFrame-3')
  })

  it('should render a Link component', () => {
    expect(authPageLayout.find(Link).exists()).toBeTruthy()
  })

  it('should render the Link component with a prop name `to`', () => {
    expect(authPageLayout.find(Link).props().to).toBe(LANDING_ROUTE)
  })

  it('should render the Link component with a prop name `className`', () => {
    expect(authPageLayout.find(Link).props().className).toBe(
      'makeStyles-flexStyles-6'
    )
  })

  it('should render Box component with a prop name `component`', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(3)
        .props().component
    ).toEqual('img')
  })

  it('should render a Box component with a prop name `src`', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(3)
        .props().src
    ).toEqual('/vectors/logo.svg')
  })

  it('should render a Box component with a prop name `alt`', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(3)
        .props().alt
    ).toEqual('logo')
  })

  it('should render a Box component with a prop name `className` at node 4', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(4)
        .props().className
    ).toBe('makeStyles-imageContainer-4')
  })

  it('should render a Box component with a prop name `component` at node 5', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(5)
        .props().component
    ).toEqual('img')
  })

  it('should render a Box component with a prop name `src` at node 5', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(5)
        .props().src
    ).toEqual('/vectors/office.svg')
  })

  it('should render a Box component with a prop name `alt` at node 5', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(5)
        .props().alt
    ).toEqual('office')
  })

  it('should render a Box component with a prop name `display` at node 5', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(5)
        .props().display
    ).toEqual('block')
  })

  it('should render a Box component with a prop name `margin` at node 5', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(5)
        .props().margin
    ).toEqual('0 auto')
  })

  it('should render a Box component with a prop name `className` at node 6', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(6)
        .props().className
    ).toBe('makeStyles-rightFrame-5')
  })

  it('should render a Box component with a prop name `className` at node 7', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(7)
        .props().className
    ).toBe('makeStyles-formContainer-7')
  })

  it('should render a Box component with a prop name `component` at node 8', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(8)
        .props().component
    ).toEqual('h2')
  })

  it('should render a Box component with a prop name `my` at node 8', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(8)
        .props().my
    ).toEqual(0)
  })

  it('should render a Box component with a prop name `className` at node 8', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(8)
        .props().className
    ).toBe('makeStyles-title-8')
  })

  it('should render a Box component with a prop name `children` at node 8', () => {
    expect(
      authPageLayout
        .find(Box)
        .at(8)
        .props().children
    ).toBeUndefined()
  })
})
