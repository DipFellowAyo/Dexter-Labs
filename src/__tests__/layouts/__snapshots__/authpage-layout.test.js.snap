// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage Layout should match the snapshot 1`] = `
<Styled(MuiBox)
  className="makeStyles-root-1"
  flexGrow="1"
>
  <Styled(MuiBox)
    className="makeStyles-gridContainer-2"
  >
    <Styled(MuiBox)
      className="makeStyles-leftFrame-3"
    >
      <Link
        className="makeStyles-flexStyles-6"
        to="/"
      >
        <Styled(MuiBox)
          alt="logo"
          component="img"
          src="/vectors/logo.svg"
        />
      </Link>
      <Styled(MuiBox)
        className="makeStyles-imageContainer-4"
      >
        <Styled(MuiBox)
          alt="office"
          component="img"
          display="block"
          margin="0 auto"
          src="/vectors/office.svg"
        />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      className="makeStyles-rightFrame-5"
    >
      <Styled(MuiBox)
        className="makeStyles-formContainer-7"
      >
        <Styled(MuiBox)
          className="makeStyles-title-8"
          component="h2"
          my={0}
        />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
